exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetGroupScan.java

```diff
@@ -819,63 +827,64 @@ private void init() throws IOException {
           }
         }
         rowGroupInfo.setEndpointByteMap(endpointByteMap);
+        rowGroupInfo.setColumns(rg.getColumns());
         rgIndex++;
         rowGroupInfos.add(rowGroupInfo);
       }
     }
 
     this.endpointAffinities = AffinityCreator.getAffinityMap(rowGroupInfos);
+    updatePartitionColTypeMap();
+  }
 
+  private void updatePartitionColTypeMap() {
     columnValueCounts = Maps.newHashMap();
     this.rowCount = 0;
     boolean first = true;
-    for (ParquetFileMetadata file : parquetTableMetadata.getFiles()) {
-      for (RowGroupMetadata rowGroup : file.getRowGroups()) {
-        long rowCount = rowGroup.getRowCount();
-        for (ColumnMetadata column : rowGroup.getColumns()) {
-          SchemaPath schemaPath = SchemaPath.getCompoundPath(column.getName());
-          Long previousCount = columnValueCounts.get(schemaPath);
-          if (previousCount != null) {
-            if (previousCount != GroupScan.NO_COLUMN_STATS) {
-              if (column.getNulls() != null) {
-                Long newCount = rowCount - column.getNulls();
-                columnValueCounts.put(schemaPath, columnValueCounts.get(schemaPath) + newCount);
-              }
-            }
-          } else {
+    for (RowGroupInfo rowGroup : this.rowGroupInfos) {
```

Comment:
Isn't this doing the same thing as the original code? RowGroupInfos is built from the RowGroupMetadata in the files?